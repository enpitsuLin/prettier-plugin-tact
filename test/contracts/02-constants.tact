import "@stdlib/deploy";

// global constants are calculated in compile-time and can't change
const GlobalConst1: Int = 1000 + ton("1.24") + pow(10, 9);

contract Constants with Deployable {

    // contract constants are calculated in compile-time and can't change
    const ContractConst1: Int = 2000 + ton("1.25") + pow(10, 9);
    
    // if your contract can be in multiple states, constants are an easy alternative to enums
    const StateUnpaid: Int = 0;
    const StatePaid: Int = 1;
    const StateDelivered: Int = 2;
    const StateDisputed: Int = 3;

    init() {}

    get fun sum(): Int {
        // you can read the constants anywhere
        return GlobalConst1 + self.ContractConst1 + self.StatePaid;
    }
}