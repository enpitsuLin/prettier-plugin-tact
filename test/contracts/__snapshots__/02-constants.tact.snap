import "@stdlib/deploy";

// global constants are calculated in compile-time and can't change

contract Constants with Deployable {
    // contract constants are calculated in compile-time and can't change

    // if your contract can be in multiple states, constants are an easy alternative to enums

    init() {}

    get fun sum(): Int {
        // you can read the constants anywhere
        return GlobalConst1 + self.ContractConst1 + self.StatePaid;
    }
}